### PR TITLE
fix!: custom object records integration tests

### PR DESCRIPTION
**Pull request by asdfryan**
This also includes some bugfixes on the relevant code:
- PATCH should return 200 not 201
- Missing required fields in hubspot should return 400 not 500

### CODE REVIEW

File: apps/api/routes/crm/v2/custom_objects/custom_object_records.integration.test.ts [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+      description__c: 'my desc here',
+      int__c: 1,
+      double__c: 0.1,
+      bool__c: true,
+    };
+  });
+
+  // describe.each(['hubspot'])('%s', (providerName) => {
+  describe.each(['hubspot', 'salesforce'])('%s', (providerName) => {
+    test(`Post /`, async () => {
```
Comment (tomkit) [inline apps/api/routes/crm/v2/custom_objects/custom_object_records.integration.test.ts:87]: this is testing quite a bit of things: can you make the test more descriptive or annotate in the test what mutations you're making and what you're testing for?

File: packages/core/remotes/impl/hubspot/index.ts [hunk 2]
```diff
@@ -2315,7 +2315,10 @@ class HubSpotClient extends AbstractCrmRemoteClient implements MarketingAutomati
 
     switch (code) {
       case 400:
-        if (message === 'one or more associations are not valid') {
+        if (
+          message === 'one or more associations are not valid' ||
+          message.includes('Some required properties were not set')
```
Comment (tomkit) [inline packages/core/remotes/impl/hubspot/index.ts:2320]: `startsWith` or `includes`?

**Comment by tomkit** [comment]
This should be a `fix!` PR not a `chore`

File: packages/core/remotes/impl/hubspot/index.ts (hunk 2 above)
Comment (asdfryan) [reply packages/core/remotes/impl/hubspot/index.ts:2320]: it says `includes` here.

File: apps/api/routes/crm/v2/custom_objects/custom_object_records.integration.test.ts (hunk 1 above)
Comment (asdfryan) [reply apps/api/routes/crm/v2/custom_objects/custom_object_records.integration.test.ts:87]: I am using the same pattern / precedent as all the other integration tests.

**Comment by asdfryan** [comment]
> This should be a `fix!` PR not a `chore`

There are no breaking changes in this PR.

File: apps/api/routes/crm/v2/custom_objects/custom_object_records.integration.test.ts (hunk 1 above)
Comment (tomkit) [reply apps/api/routes/crm/v2/custom_objects/custom_object_records.integration.test.ts:87]: Can we improve upon it for readability; I haven't gotten to reviewing every PR so some patterns we can improve upon

**Comment by tomkit** [comment]
> > This should be a `fix!` PR not a `chore`
> 
> There are no breaking changes in this PR.

The response code change is breaking

File: packages/core/remotes/impl/hubspot/index.ts (hunk 2 above)
Comment (tomkit) [reply packages/core/remotes/impl/hubspot/index.ts:2320]: I'm more asking what it should be: it looks like we're expecting it to be the start of the message, and the end of the message is dynamic?

Comment (asdfryan) [reply packages/core/remotes/impl/hubspot/index.ts:2320]: If you look at the integration test I added, you can see that the full error message is `Error creating {{CUSTOM_OBJ_NAME}}. Some required properties were not set.`

File: apps/api/routes/crm/v2/custom_objects/custom_object_records.integration.test.ts (hunk 1 above)
Comment (asdfryan) [reply apps/api/routes/crm/v2/custom_objects/custom_object_records.integration.test.ts:87]: In that case I would like to open a separate PR where I can address this holistically across all of our integration tests. Does that sound good to you?

**Review by tomkit — APPROVED** [review]

**Pull request merged**